src/linear_algebra/general_linear_group.lean

```diff
@@ -0,0 +1,169 @@
+/-
+Copyright (c) 2021 Chris Birkbeck. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Birkbeck
+-/
+import linear_algebra.matrix
+import linear_algebra.matrix.nonsingular_inverse
+import linear_algebra.special_linear_group
+import linear_algebra.determinant
+
+/-!
+# The General Linear group $GL(n, R)$
+
+This file defines the elements of the General Linear group `general_linear_group n R`,
+consisting of all invertible `n` by `n` `R`-matrices.
+
+
+## Main definitions
+
+ * `matrix.general_linear_group` is the type of matrices over R which are units in the matrix ring.
+ * `matrix.GL_plus.GL_pos` gives the subgroup of matrices with
+ positive determinant (over a linear ordered ring)
+
+## Implementation notes
+
+
+## References
+
+
+## Tags
+
+matrix group, group, matrix inverse
+-/
+
+namespace matrix
+universes u v
+open_locale matrix
+open linear_map
+
+namespace general_linear_group
+
+variables {n : Type u} [decidable_eq n] [fintype n] {R : Type v} [comm_ring R]
+
+/-- `GL n R` is the group of `n` by `n` `R`-matrices with unit determinant.
+Defined as a subtype of matrices-/
+abbreviation general_linear_group (n : Type u) (R : Type v)
+  [decidable_eq n] [fintype n] [comm_ring R] : Type* := units (matrix n n R)
```

Comment:
This definition should live outside the namespace.

Reply:
Aha fixed. Is there a good way to do this for GL_pos also? At the moment I put it in the GL_plus namespace but I'm not sure if thats optimal. I guess it doesnt need to be in a namespace?

Reply:
Is there a reason you're using both the names GL_plus and GL_pos?

Reply:
Originally I had just GL_pos, but my lint said not to use that for the namespace. I thought maybe this should be in a name space, but maybe because its the definition is doesn't need to be?